ts/features/itwallet/common/store/reducers/index.ts

```diff
@@ -43,15 +45,28 @@ const itwReducer = combineReducers({
   preferences: preferencesReducer
 });
 
-const CURRENT_REDUX_ITW_STORE_VERSION = 1;
+const CURRENT_REDUX_ITW_STORE_VERSION = 2;
 
 const migrations: MigrationManifest = {
   // Added preferences store
   "0": (state: PersistedState): PersistedState =>
     _.set(state, "preferences", {}),
+
   // Added requestedCredentials to preferences store
   "1": (state: PersistedState): PersistedState =>
-    _.set(state, "preferences.requestedCredentials", {})
+    _.set(state, "preferences.requestedCredentials", {}),
+
+  // Added authLevel to preferences store and set it to "L2" if eid is present
+  "2": (state: PersistedState): PersistedState => {
+    const eid: O.Option<StoredCredential> = _.get(state, "credentials.eid");
+
+    // If eid is a Some(value), set authLevel to "L2"
+    if (eid && O.isSome(eid)) {
+      return _.set(state, "preferences.authLevel", "L2");
```

Comment:
[`lodash.set` mutates the object passed by parameter](https://lodash.com/docs/4.17.15#set), which is strongly discouraged.

Reply:
Thanks for pointing it out!

My alternative solution without using mutation `(_.set)` is this one, but I know that this is not type-safe since we bypass strict TypeScript checks. What do you think? Do you have any other type-safe solution?

```
  // Added authLevel to preferences store and set it to "L2" if eid is present
  "2": (state: PersistedState): PersistedState => {
    const eid: O.Option<StoredCredential> = (state as any).credentials?.eid;

    // If there is an eid, set the authLevel to "L2"
    if (eid && O.isSome(eid)) {
      return {
        ...state,
        preferences: {
          ...((state as any).preferences || {}),
          authLevel: "L2"
        }
      } as PersistedState;
    }

    return state;
  }
```

Reply:
Yes I think it should be ok, we could simplify it more like this:
```typescript
// Added authLevel to preferences store and set it to "L2" if eid is present
  "2": (state: PersistedState): PersistedState => {
    const { credentials, preferences } = state as PersistedItWalletState;

    // If eid is a Some(value), set authLevel to "L2"
    if (O.isSome(credentials.eid)) {
      return {
        ...state,
        preferences: {
          ...preferences,
          authLevel: "L2"
        }
      } as PersistedItWalletState;
    }

    return state;
  }
  ```